### PR TITLE
Retrieve credentials from instance profile

### PR DESCRIPTION
**Pull request by shaun7pan**
What:
Retrieve credentials from instance profile.
[1]: https://github.com/concourse/s3-resource/pull/22

How to review:
Run test using `./script/test`

### CODE REVIEW

File: s3client.go [hunk 1]
```diff
@@ -117,7 +119,17 @@ func NewAwsConfig(
 	var creds *credentials.Credentials
 
 	if accessKey == "" && secretKey == "" {
-		creds = credentials.AnonymousCredentials
+		// Try with instance profile
```
Comment (combor) [inline s3client.go]: I don't think we need any of this here. 

From the docs: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html
When you initialize a new service client without providing any credential arguments, the SDK uses the default credential provider chain to find AWS credentials. The SDK uses the first provider in the chain that returns credentials without an error. The default provider chain looks for credentials in the following order:
- Environment variables.
- Shared credentials file.
- If your application uses an ECS task definition or RunTask API operation, IAM role for tasks.
- If your application is running on an Amazon EC2 instance, IAM role for Amazon EC2.

The SDK detects and uses the built-in providers automatically, without requiring manual configurations. For example, if you use IAM roles for Amazon EC2 instances, your applications automatically use the instance’s credentials. You don’t need to manually configure credentials in your application.

So basically we can delete any credential check and simply initialize a session like this:
```
 sess := session.Must(session.NewSession())
 ```
and delete any other code that tries to grab credentials.

**Review by combor — CHANGES_REQUESTED** [review]

File: s3client.go (hunk 1 above)
Comment (shaun7pan) [reply s3client.go]: All the agrees. Changed to use aws go sdk default behavior when creating new session.
Please check again.
